nexus/src/external_api/http_entrypoints.rs

```diff
@@ -5048,6 +5058,201 @@ async fn updates_refresh(
+    let handler = async {
+        let opctx = OpContext::for_external_api(&rqctx).await?;
+        let components = nexus
+            .updateable_components_list_by_id(&opctx, &pagparams)
+            .await?
+            .into_iter()
+            .map(|u| u.into())
+            .collect();
+        Ok(HttpResponseOk(ScanById::results_page(
+            &query,
+            components,
+            &|_, u: &UpdateableComponent| u.identity.id,
+        )?))
+    };
+    apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
+}
+
+/// List all updates
+#[endpoint {
+     method = GET,
+     path = "/v1/system/update/updates",
+     tags = ["system"],
+}]
+async fn system_update_list(
+    rqctx: Arc<RequestContext<Arc<ServerContext>>>,
+    // TODO: pagination should probably be by most recent first, ID is nonsense
+    query_params: Query<PaginatedById>,
+) -> Result<HttpResponseOk<ResultsPage<SystemUpdate>>, HttpError> {
+    let apictx = rqctx.context();
+    let nexus = &apictx.nexus;
+    let query = query_params.into_inner();
+    let pagparams = data_page_params_for(&rqctx, &query)?;
+    let handler = async {
+        let opctx = OpContext::for_external_api(&rqctx).await?;
+        let updates = nexus
+            .system_updates_list_by_id(&opctx, &pagparams)
+            .await?
+            .into_iter()
+            .map(|u| u.into())
+            .collect();
+        Ok(HttpResponseOk(ScanById::results_page(
+            &query,
+            updates,
+            &|_, u: &SystemUpdate| u.identity.id,
+        )?))
+    };
+    apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
+}
+
+/// Path parameters for SystemUpdate requests
+#[derive(Deserialize, JsonSchema)]
+struct SystemUpdatePathParam {
+    // TODO: do updates have names? Should this therefore be NameOrId? Do we
+    // allow access by some other string identifier (like version) which, like
+    // Name, is disjoint with the set of Uuids? NameOrVersion?
+    id: Uuid,
+}
+
+/// View system update
+#[endpoint {
+     method = GET,
+     path = "/v1/system/update/updates/{id}",
+     tags = ["system"],
+}]
+async fn system_update_view(
+    rqctx: Arc<RequestContext<Arc<ServerContext>>>,
+    path_params: Path<SystemUpdatePathParam>,
+) -> Result<HttpResponseOk<SystemUpdate>, HttpError> {
+    let apictx = rqctx.context();
+    let nexus = &apictx.nexus;
+    let path = path_params.into_inner();
+    let handler = async {
+        let opctx = OpContext::for_external_api(&rqctx).await?;
+        let system_update =
+            nexus.system_update_fetch_by_id(&opctx, &path.id).await?;
+        Ok(HttpResponseOk(system_update.into()))
+    };
+    apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
+}
+
+/// View system update component tree
+#[endpoint {
+    method = GET,
+    path = "/v1/system/update/updates/{id}/components",
+    tags = ["system"],
+}]
+async fn system_update_components_list(
```

Comment:
So we settled on not pagination the component lists? I'd saw that mentioned in the RFD.

Reply:
Yes, for now, but it's easy to change because it's its own endpoint and it already returns a `ResultsPage`.